packages/fiori-elements-writer/src/index.ts

```diff
@@ -64,13 +76,31 @@ async function generate<T>(basePath: string, data: FioriElementsApp<T>, fs?: Edi
         JSON.parse(render(fs.read(join(rootTemplatesPath, 'common', 'extend', 'package.json')), feApp, {}))
     );
 
-    const templateVersionPath = join(rootTemplatesPath, `v${feApp.service?.version}`);
-    // Copy version specific common templates and version specific, floorplan specific templates
-    [join(templateVersionPath, 'common', 'add'), join(templateVersionPath, feApp.template.type, 'add')].forEach(
-        (templatePath) => {
-            fs!.copyTpl(join(templatePath, '**/*.*'), basePath, feApp, {}, { ignoreNoMatch: true });
+    // Special handling for FPM because it is not based on template files but used the fpm writer
+    if (feApp.template.type === TemplateType.FlexibleProgrammingModel) {
+        if (feApp.service?.version === OdataVersion.v2) {
```

Comment:
Im not sure this is necessary as we already have validation for service version here: https://github.com/SAP/open-ux-tools/pull/626/files#diff-a51a19d6fa0b94939e84e2a4eedd1b602cae40c2077c22ecce59ee537f5ccbcfR40 . Can we use the same approach for the new template also?

Reply:
Thanks, I have removed the check and added a test case validating it with https://github.com/SAP/open-ux-tools/pull/626/commits/56127e442f97065ab01e0d4568863ef3d8f18e6b